ogzaf/demographics.py

```diff
@@ -104,83 +360,59 @@ def get_fert(totpers, min_yr, max_yr, graph=False):
-        )
-        / 2000
-    )
-    age_midp = np.array([9, 12, 15, 17, 22, 27, 32, 37, 42, 47, 52, 57])
-    # Generate interpolation functions for fertility rates
-    fert_func = si.interp1d(age_midp, fert_data, kind="cubic")
-    # Calculate average fertility rate in each age bin using trapezoid
-    # method with a large number of points in each bin.
-    binsize = (max_yr - min_yr + 1) / totpers
-    num_sub_bins = float(10000)
-    len_subbins = (np.float64(100 * num_sub_bins)) / totpers
-    age_sub = (
-        np.linspace(
-            np.float64(binsize) / num_sub_bins,
-            np.float64(max_yr),
-            int(num_sub_bins * max_yr),
+
+    # Extrapolate fertility rates for ages 1-14 and 50-100 using exponential
+    # function
+    ages_1_14 = np.arange(1, 15)
+    slope_15 = (fert_rates_15_49[1] - fert_rates_15_49[0]) / (
+        ages_15_49[1] - ages_15_49[0]
+    )
+    fert_rates_1_14 = extrap_exp_3(
+        ages_1_14, (15, fert_rates_15_49[0]), slope_15, (9, 0.0001), low=True
+    )
+    ages_50_100 = np.arange(50, 101)
+    slope_49 = (fert_rates_15_49[-1] - fert_rates_15_49[-2]) / (
+        ages_15_49[-1] - ages_15_49[-2]
+    )
+    fert_rates_50_100 = extrap_exp_3(
+        ages_50_100,
+        (49, fert_rates_15_49[-1]),
+        slope_49,
+        (57, 0.0001),
+        low=False,
+    )
+    fert_rates = np.hstack(
+        (fert_rates_1_14, fert_rates_15_49, fert_rates_50_100)
+    )
+    ages = np.arange(1, 101)
+
+    if graph:  # Plot fertility rates
+        plt.plot(ages, fert_rates)
+        plt.xlabel(r"Age $s$")
+        plt.ylabel(r"Fertility rate $f_{s}$")
+        plt.text(
+            -5,
+            -0.023,
+            "Source: UN Population Data",
+            fontsize=9,
         )
-        - 0.5 * np.float64(binsize) / num_sub_bins
-    )
-    curr_pop_sub = np.repeat(
-        np.float64(curr_pop_pct) / num_sub_bins, num_sub_bins
-    )
-    fert_rates_sub = np.zeros(curr_pop_sub.shape)
-    pred_ind = (age_sub > age_midp[0]) * (age_sub < age_midp[-1])
-    age_pred = age_sub[pred_ind]
-    fert_rates_sub[pred_ind] = np.float64(fert_func(age_pred))
-    fert_rates = np.zeros(totpers)
-    end_sub_bin = 0
-    for i in range(totpers):
-        beg_sub_bin = int(end_sub_bin)
-        end_sub_bin = int(np.rint((i + 1) * len_subbins))
-        fert_rates[i] = (
-            curr_pop_sub[beg_sub_bin:end_sub_bin]
-            * fert_rates_sub[beg_sub_bin:end_sub_bin]
-        ).sum() / curr_pop_sub[beg_sub_bin:end_sub_bin].sum()
-
-    # if graph:  # need to fix plot function for new data output
-    #     pp.plot_fert_rates(fert_rates, age_midp, totpers, min_yr, max_yr,
-    #                        fert_data, fert_rates, output_dir=OUTPUT_DIR)
+        plt.tight_layout(rect=(0, 0.035, 1, 1))
+        output_path = os.path.join(OUTPUT_DIR, "fert_rates")
+        plt.savefig(output_path)
+        plt.close()
 
     return fert_rates
 
 
 def get_mort(totpers, min_yr, max_yr, graph=False):
```

Comment:
```suggestion
def get_mort(min_yr, max_yr, graph=False):
```

Reply:
@SeaCelo. I need to work on this a little bit. We discussed hard coding the `totpers` object to be `100` and `min_yr=1` and `max_yr=100`. But I need to look back at all the tests and `calibrate.py` that use this to make sure this is OK. We might need it to be flexible.

Reply:
@SeaCelo. I removed the `min_yr` and `max_yr` arguments in `demographics.py` and `calibrate.py` as they are hard coded to `min_yr=1` and `max_yr=100` because the UN population data have annual ages. I left the `totpers` argument because we need to be able to transform the annual data to model periods that are bigger than one year (`totpers<100`).